### PR TITLE
Added functionality: dropout in either train, test, or both phases

### PR DESCRIPTION
**Pull request by karimpedia**
A flag added to the Dropout class that determines in which phase[s] the dropout takes place (either train, test, or both phases).

### CODE REVIEW

File: keras/layers/core.py [hunk 1]
```diff
@@ -90,10 +90,11 @@ class Dropout(Layer):
         - [Dropout: A Simple Way to Prevent Neural Networks from Overfitting](http://www.cs.toronto.edu/~rsalakhu/papers/srivastava14a.pdf)
     """
 
-    def __init__(self, p, noise_shape=None, seed=None, **kwargs):
+    def __init__(self, p, noise_shape=None, seed=None, active_in='train', **kwargs):
```
Comment (Dref360) [inline keras/layers/core.py]: Update doc to match new args.
Also, what,s the use case of this?

Comment (karimpedia) [inline keras/layers/core.py]: Updated !

Dropout in inference phase is a [proposed] way to introduce stochasticity in a model.

For instance: Isola et. al [https://arxiv.org/pdf/1611.07004v1.pdf] use this technique in order to introduce some randomness in the generator model of a conditional Generative Adversarial Network. They, however, report that the effect at the output is minor.

**Comment by fchollet** [comment]
We will take care of this functionality in Keras, by making it possible to set the learning phase on a per-layer basis.
